joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForPrimitivesCreator.scala

```diff
@@ -1,23 +1,26 @@
 package io.joern.csharpsrc2cpg.astcreation
 
+import io.joern.csharpsrc2cpg.datastructures.FieldDecl
 import io.joern.csharpsrc2cpg.parser.DotNetJsonAst.*
 import io.joern.csharpsrc2cpg.parser.{DotNetNodeInfo, ParserKeys}
 import io.joern.x2cpg.{Ast, Defines, ValidationMode}
-import io.shiftleft.codepropertygraph.generated.nodes.{NewCall, NewLocal}
+import io.shiftleft.codepropertygraph.generated.nodes.{DeclarationNew, NewCall, NewLocal}
 import io.shiftleft.codepropertygraph.generated.{DispatchTypes, Operators}
 
 trait AstForPrimitivesCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
 
   protected def astForIdentifier(ident: DotNetNodeInfo, typeFullName: String = Defines.Any): Ast = {
     val identifierName = nameFromNode(ident)
     if identifierName != "_" then {
-      val variableOption = scope.lookupVariable(identifierName)
-      variableOption match
-        case Some(variable) =>
+      List(scope.lookupVariable(identifierName), scope.findFieldInScope(identifierName)) match {
+        case List(Some(variable: DeclarationNew), None) =>
           val node = identifierFromDecl(variable, Option(ident))
           Ast(node).withRefEdge(node, variable)
+        case List(None, Some(field: FieldDecl)) =>
+          Ast(identifierNode(ident, field.name, ident.code, field.typeFullName))
```

Comment:
If this is a field, shouldn't it rather be a field access + field identifier? @AndreiDreyer has been working on field accesses and may be able to advise.

Reply:
I've seen cases where a `member` is accessed just by it's name, do we want to create a `fieldAccess` node here as well?

Reply:
When a local variable is overshadowing a field then the local is used, but if it's referring to the field, then a field access should be used. See this test case for an example: https://github.com/joernio/joern/blob/03e0ba64e3800760d59db9c72057a9c1242b99e5/joern-cli/frontends/csharpsrc2cpg/src/test/scala/io/joern/csharpsrc2cpg/querying/ast/MemberTests.scala#L273